ibis/expr/datatypes.py

```diff
@@ -1076,14 +1136,14 @@ def from_string(value):
 
 
 @dtype.register(list)
-def from_list(values):
+def from_list(values: List[GenericAny]) -> Array:
```

Comment:
```suggestion
def from_list(values: List[Union[DataType, str]]) -> Array:
```

Reply:
Actually, we're trying to figure out what the dtype is of a list based on the _values_ it contains rather than `DataType`s or `str` representations of `DataType`s, so we can't actually use this.